rig-milvus/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+use reqwest::StatusCode;
+use rig::{
+    Embed, OneOrMany,
+    embeddings::{Embedding, EmbeddingModel},
+    vector_store::{VectorStoreError, VectorStoreIndex},
+};
+use serde::{Deserialize, Serialize};
+
+/// Represents a vector store implementation using Milvus - <https://milvus.io/> as the backend.
+pub struct MilvusVectorStore<M> {
+    /// Model used to generate embeddings for the vector store
+    model: M,
```

Comment:
This might be opinionated but, is it better to keep the bounds of `M` on the struct itself instead of the method. like, theoretically, you could create a vector store given a model `M` that isn't bound my `EmbeddingModel` but you wouldn't be able to do anything with it.

I think i've seen both ways in the wild (rig-surrealdb seems to restrict the struct via generic bounds).